src/Illuminate/Database/Eloquent/Factories/Factory.php

```diff
@@ -370,9 +370,9 @@ protected function getExpandedAttributes(?Model $parent)
     protected function getRawAttributes(?Model $parent)
     {
         return $this->states->pipe(function ($states) {
-            return $this->for->isEmpty() ? $states : new Collection(array_merge([function () {
+            return $this->for->isEmpty() ? $states : new Collection(array_merge($states->all(), [function () {
                 return $this->parentResolvers();
-            }], $states->all()));
+            }]));
```

Comment:
A bit worried this is going to have more breaking change issues than we are anticipating.

Reply:
@taylorotwell I was worried about that as well, and you probably understand the potential implications more than I do. I will say this, I made this change on my side project which has 541 tests that all use the new factory system. This change didn't break any of the existing tests. Admittedly anecdotal, but it made me feel more confident about it.